src/components/WalletSelectionComponent.jsx

```diff
@@ -89,24 +87,35 @@ class WalletSelectionComponent extends Component {
           </Grid>
         </Grid>
       )
-    }
-
-    return (
-      <Grid container direction='row' alignItems='center'>
-        <Grid item>
-          <CheckIcon className={classes.connectedIcon} />
+    } else if (actionsPending.syncAccountInfo) {
+      return (
+        <Grid container direction='column' justify='center' alignItems='center'>
+          <Grid item>
+            <CircularProgress className={classes.circularProgress} />
+          </Grid>
+          <Grid item>
+            <Typography className={classes.connectedtext}>Synchronizing Acoount Info</Typography>
+          </Grid>
         </Grid>
-        <Grid item>
-          <Typography className={classes.connectedtext}>Account retrieved, your balance: {
-            numeral(utils.toHumanReadableUnit(wallet.accounts[0].balance[cryptoType])).format('0.000a')} {getCryptoSymbol(cryptoType)}
-          </Typography>
+      )
+    } else if (wallet && wallet.crypto && wallet.crypto[cryptoType] && wallet.crypto[cryptoType][0] && !actionsPending.syncAccountInfo) {
```

Comment:
wallet.crypto is never null or undefined, see https://github.com/ventureum/Chainsfer/blob/fa1cfab611a818c0f368af9dea63361ea0fed612/src/reducers/walletReducer.js#L24